dvc/repo/get.py

```diff
@@ -21,63 +20,38 @@ def __init__(self):
 
 
 def get(url, path, out=None, rev=None, jobs=None, force=False):
-    import shortuuid
-
     from dvc.dvcfile import is_valid_filename
-    from dvc.external_repo import external_repo
     from dvc.fs.callbacks import Callback
+    from dvc.repo import Repo
 
     out = resolve_output(path, out, force=force)
 
     if is_valid_filename(out):
         raise GetDVCFileError()
 
-    # Creating a directory right beside the output to make sure that they
-    # are on the same filesystem, so we could take the advantage of
-    # reflink and/or hardlink. Not using tempfile.TemporaryDirectory
-    # because it will create a symlink to tmpfs, which defeats the purpose
-    # and won't work with reflink/hardlink.
-    dpath = os.path.dirname(os.path.abspath(out))
-    tmp_dir = os.path.join(dpath, "." + str(shortuuid.uuid()))
-
-    # Try any links possible to avoid data duplication.
-    #
-    # Not using symlink, because we need to remove cache after we
-    # are done, and to make that work we would have to copy data
-    # over anyway before removing the cache, so we might just copy
-    # it right away.
-    #
-    # Also, we can't use theoretical "move" link type here, because
-    # the same cache file might be used a few times in a directory.
-    cache_types = ["reflink", "hardlink", "copy"]
```

Comment:
with `dvcfs` we just download stuff into place, there is no `fetch` involved like it used to in the past. So we can finally drop this obscure logic 🤷‍♂️